Reload triggers remote file installation documentation (#452)
cylc/cylc-doc: src/glossary.rst

      workflow :term:`source directory` before reload, rather than made by
      editing the installed files directly.

>       :ref:`RemoteInit` will be triggered upon reload. This will not take effect
>       for running tasks but for the job submission of subsequent tasks.

Reviewer: Not really triggered until the first job submits to the remote platform?

```suggestion
      :ref:`RemoteInit` will be redone for each job platform, when the first job is submitted there after a reload.
```
